cat_indices.go

```diff
@@ -240,135 +256,176 @@ type CatIndicesResponse []CatIndicesResponseRow
+	WarmerTotalTime              string `json:"warmer.total_time" alias:"wtt,warmerTotalTime"`                       // time spent in warmers on primaries & replicas, e.g. "47s"
+	PriWarmerTotalTime           string `json:"pri.warmer.total_time"`                                               // time spent in warmers on primaries, e.g. "47s"
+	SuggestCurrent               int    `json:"suggest.current,string" alias:"suc,suggestCurrent"`                   // number of current suggest ops on primaries & replicas
+	PriSuggestCurrent            int    `json:"pri.suggest.current,string"`                                          // number of current suggest ops on primaries
+	SuggestTime                  string `json:"suggest.time" alias:"suti,suggestTime"`                               // time spend in suggest on primaries & replicas, "31s"
+	PriSuggestTime               string `json:"pri.suggest.time"`                                                    // time spend in suggest on primaries, e.g. "31s"
+	SuggestTotal                 int    `json:"suggest.total,string" alias:"suto,suggestTotal"`                      // number of suggest ops on primaries & replicas
+	PriSuggestTotal              int    `json:"pri.suggest.total,string"`                                            // number of suggest ops on primaries
+	MemoryTotal                  string `json:"memory.total" alias:"tm,memoryTotal"`                                 // total user memory on primaries & replicas, e.g. "1.5kb"
+	PriMemoryTotal               string `json:"pri.memory.total"`                                                    // total user memory on primaries, e.g. "1.5kb"
+}
+
+// catIndicesResponseRowAliasesMap holds the global compiled map for columns aliases
+// the map is used by CatIndicesService.buildURL
+var catIndicesResponseRowAliasesMap map[string]string
+
+// catIndicesResponseRowAliasesMap needs reflect to be built
+// but it is used once only on the the application start (so considered not harmful for performance)
+func init() {
+	catIndicesResponseRowAliasesMap = make(map[string]string)
+
+	valType := reflect.TypeOf(CatIndicesResponseRow{})
+
+	for i := 0; i < valType.NumField(); i++ {
+		field := valType.Field(i)
+
+		jsonVal := field.Tag.Get("json")
+		if jsonVal == "" {
+			continue
+		}
+
+		jsonVal = strings.Split(jsonVal, ",")[0]
+
+		aliasVal := field.Tag.Get("alias")
+		if aliasVal == "" {
+			continue
+		}
+
+		// for backwards compatibility some fields are able to have the same aliases
+		// that means that one alias can be translated to different columns (from different elastic versions)
+		// example for understanding: rto -> RefreshTotal, RefreshExternalTotal
+		for _, alias := range strings.Split(aliasVal, ",") {
+			if existed, ok := catIndicesResponseRowAliasesMap[alias]; ok {
+				catIndicesResponseRowAliasesMap[alias] = existed + "," + jsonVal
+			} else {
+				catIndicesResponseRowAliasesMap[alias] = jsonVal
+			}
+		}
+	}
+
+	return
```

Comment:
Although this works, I think it's overkill to use reflection for it. Can't we just use a simple map?

```
var catIndicesResponseRowAliasesMap = map[string]string{
  "i":   "index",
  "idx": "index",
  ...
}
```

Keep it simple and stupid.

Reply:
Yeap, as I described above - I understand that reflect maybe a complication, so I offer two options: reflect and a simple hardcoded map.

The only drawback of a map, as I see, is - more possible case that it can be out-to-date when the main struct is updated (so e.g. a field is added/removed and the alias was not updated, as it is stored in a separate map). 

If you're OK with it too, I'll switch to a single map then

Reply:
Yeah. We simply add it then. It's just a simple bug. You could've forgotten the alias as well.